addons/website/controllers/main.py

```diff
@@ -835,56 +835,112 @@ def seo_suggest(self, keywords=None, lang=None):
         xmlroot = ET.fromstring(response)
         return json.dumps([sugg[0].attrib['data'] for sugg in xmlroot if len(sugg) and sugg[0].attrib['data']])
 
-    @http.route(['/website/get_alt_images'], type='jsonrpc', auth="user", website=True)
-    def get_alt_images(self, models):
+    def extract_thumbnail_url(self, src):
+        if "youtube.com" in src or "youtu.be" in src:
+            video_id = re.search(r"(?:embed/|youtu\.be/)([^?&/]+)", src)
+            if video_id:
+                return f"https://img.youtube.com/vi/{video_id.group(1)}/maxresdefault.jpg"
+        elif "vimeo.com" in src:
+            video_id = re.search(r"/video/(\d+)", src)
+            if video_id:
+                return f"https://vumbnail.com/{video_id.group(1)}.jpg"
+        elif "dailymotion.com" in src:
+            video_id = re.search(r"/video/([^?&/]+)", src)
+            if video_id:
+                return f"https://www.dailymotion.com/thumbnail/video/{video_id.group(1)}"
```

Comment:
In Python, use` ' ` for technical strings, ` " `is only for human-readable text. Please update everywhere in python files.